File: easy/944_Delete_Columns_to_Make_Sorted.py
def minDeletionSize(strs):
    count_delete = 0
    n = len(strs)
    for i in range(len(strs[0])):
        column = ''
        for j in range(n):
            column += strs[j][i]
        print(column)
        if column != sorted("".join(column)):
            count_delete += 1
    return count_delete

print(minDeletionSize(["abc", "bce", "cae"]))

s = 'abcd'
d = ''.join(sorted(s))
print(d)
print(s == d)
        

File: easy/944_Delete_Columns_to_Make_Sorted.cpp
class Solution {
public:
    int minDeletionSize(vector<string>& strs) {
        int delete_count = 0;
        int n = size(strs);

        for (int i=0; i < size(strs[0]); i++){
            for (int j=0; j < n-1; j++){
                if (strs[j][i] > strs[j+1][i]){
                    delete_count++;
                    break;
                }
            }
        }

        return delete_count;

    }
};